rpc/src/helpers.rs

```diff
@@ -244,6 +361,75 @@ pub struct RpcErrorMsg {
 //     }
 // }
 
+#[derive(Serialize, Debug, Clone)]
+pub struct Protocols {
+    protocol: String,
+    next_protocol: String,
+}
+
+impl Protocols {
+    pub fn new(protocol: String, next_protocol: String) -> Self {
+        Self {
+            protocol,
+            next_protocol,
+        }
+    }
+}
+
+// ---------------------------------------------------------------------
+#[derive(Serialize, Debug, Clone)]
+pub struct NodeVersion {
+    version: Version,
+    network_version: NetworkVersion,
+    commit_info: CommitInfo,
+}
+
+#[derive(Serialize, Debug, Clone)]
+pub struct NetworkVersion {
+    chain_name: String,
+    distributed_db_version: i32,
+    p2p_version: i32,
+}
+
+#[derive(Serialize, Debug, Clone)]
+pub struct CommitInfo {
+    commit_hash: UniString,
+    commit_date: UniString,
+}
+
+#[derive(Serialize, Debug, Clone)]
+pub struct Version {
+    major: i32,
+    minor: i32,
+    additional_info: String,
+}
+
+impl NodeVersion {
+    pub fn new(env_config: &TezosEnvironmentConfiguration) -> Self {
+        let version_env: &'static str = env!("CARGO_PKG_VERSION");
+
+        let version: Vec<String> = version_env.split(".").map(|v| v.to_string()).collect();
+
+        Self {
+            version: Version {
+                major: version[0].parse().unwrap_or(0),
+                minor: version[1].parse().unwrap_or(0),
+                additional_info: "release".to_string(),
+            },
+            network_version: NetworkVersion {
+                chain_name: env_config.version.clone(),
+                distributed_db_version: 0,
```

Comment:
@adonagy 
we have constants for these versions:
```
networking\src\p2p\peer.rs

const SUPPORTED_DISTRIBUTED_DB_VERSION: u16 = 0;
const SUPPORTED_P2P_VERSION: u16 = 1;
```
Should be better to use them here.
So, I would move those constants to:
networking\src\p2p\mod.rs or networking\src\lib.rs